emails/views.py

```diff
@@ -76,15 +77,20 @@ def _index_POST(request):
 
     incr_if_enabled('emails_index_post', 1)
     existing_addresses = RelayAddress.objects.filter(user=user_profile.user)
-    if existing_addresses.count() >= settings.MAX_NUM_BETA_ALIASES:
-        if 'moz-extension' in request.headers.get('Origin', ''):
-            return HttpResponse('Payment Required', status=402)
-        messages.error(
-            request, "You already have 5 email addresses. Please upgrade."
+
+    with transaction.atomic():
+        profile = Profile.objects.select_for_update().get(
+            user=user_profile.user
         )
```

Comment:
The user profile is already available from line 72's assignment of `user_profile`.
```suggestion
```